Loop/Managers/LoopDataManager.swift

```diff
@@ -147,11 +221,23 @@ class LoopDataManager {
             }
         }
 
+        if insulinOnBoard == nil {
+            dispatch_group_enter(updateGroup)
+            deviceDataManager.doseStore.insulinOnBoardAtDate(NSDate()) { (value, error) in
```

Comment:
from a dosing perspective, yeah, you'd want it to be as close-as-possible to `startDate` on line 386, which it will be here, since these data retrievals should be pretty fast. If it needs to be exact, you could set `startDate` on line 186 and pass it in as an argument to `self.updatePredictedGlucoseAndRecommendedBasal`